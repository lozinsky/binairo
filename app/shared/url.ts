export function resolvePathname(pathname: string) {
  return pathname.replace(/^\//, import.meta.env.BASE_URL);
}
