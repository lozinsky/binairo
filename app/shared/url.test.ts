import { expect, test, vi } from 'vitest';

import { resolvePathname } from './url';

test.each([
  ['/settings', '/'],
  ['/settings', '/binairo/'],
  ['/game/:board', '/'],
  ['/game/:board', '/binairo/'],
])('resolves "%s" pathname if `BASE_URL` environment variable is "%s"', (actual, baseUrl) => {
  vi.stubEnv('BASE_URL', baseUrl);

  expect(resolvePathname(actual)).toMatchSnapshot();
});
