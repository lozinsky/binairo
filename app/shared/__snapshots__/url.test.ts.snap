// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`resolves "/game/:board" pathname if \`BASE_URL\` environment variable is "/" 1`] = `"/game/:board"`;

exports[`resolves "/game/:board" pathname if \`BASE_URL\` environment variable is "/binairo/" 1`] = `"/binairo/game/:board"`;

exports[`resolves "/settings" pathname if \`BASE_URL\` environment variable is "/" 1`] = `"/settings"`;

exports[`resolves "/settings" pathname if \`BASE_URL\` environment variable is "/binairo/" 1`] = `"/binairo/settings"`;
